src/core/webidl.js

```diff
@@ -5,181 +5,112 @@
-  ["Uint32Array", "WEBIDL#idl-Uint32Array"],
-  ["Uint8Array", "WEBIDL#idl-Uint8Array"],
-  ["Uint8ClampedArray", "WEBIDL#dl-Uint8ClampedArray"],
-  ["unrestricted double", "WEBIDL#idl-unrestricted-double"],
-  ["unrestricted float", "WEBIDL#idl-unrestricted-float"],
-  ["unsigned long long", "WEBIDL#idl-unsigned-long-long"],
-  ["unsigned long", "WEBIDL#idl-unsigned-long"],
-  ["unsigned short", "WEBIDL#idl-unsigned-short"],
-  ["USVString", "WEBIDL#idl-USVString"],
-]);
-
 const operationNames = {};
 const idlPartials = {};
 
-// Takes the result of WebIDL2.parse(), an array of definitions.
-function makeMarkup(parse, { suppressWarnings } = {}) {
-  const templates = {
-    wrap(items) {
-      return items
-        .reduce(flatten, [])
-        .map(x => (typeof x === "string" ? new Text(x) : x));
-    },
-    trivia(t) {
-      if (!t.trim()) {
-        return t;
-      }
-      return hyperHTML`<span class='idlSectionComment'>${t}</span>`;
-    },
-    generic(wrapped) {
-      if (standardTypes.has(wrapped)) {
-        return hyperHTML`<a data-cite='${standardTypes.get(
-          wrapped
-        )}'>${wrapped}</a>`;
-      }
-      return hyperHTML`<a data-link-for="">${wrapped}</a>`;
-    },
-    reference(wrapped, name) {
-      if (standardTypes.has(name)) {
-        return hyperHTML`<a data-cite='${standardTypes.get(
-          name
-        )}'>${wrapped}</a>`;
-      }
-      return hyperHTML`<a data-link-for="">${wrapped}</a>`;
-    },
-    name(escaped, { data, parent }) {
-      if (data.idlType && data.idlType.type === "argument-type") {
-        return hyperHTML`<span class="idlParamName">${escaped}</span>`;
-      }
-      const parentName = parent ? parent.name : "";
-      const { name } = getNameAndId(data, parentName);
-      const dfn = findDfn(data, name, {
-        parent: parentName,
-        suppressWarnings,
-      });
-      const idlAnchor = createIdlAnchor(escaped, data, parentName, dfn);
-      const className = parent ? "idlName" : "idlID";
-      if (data.type === "enum-value") {
-        return idlAnchor;
-      }
-      return hyperHTML`<span class="${className}">${idlAnchor}</span>`;
-    },
-    type: contents => hyperHTML`<span class="idlType">${contents}</span>`,
-    inheritance: contents =>
-      hyperHTML`<span class="idlSuperclass">${contents}</span>`,
-    definition: createIdlElement,
-    extendedAttribute: contents =>
-      hyperHTML`<span class="extAttr">${contents}</span>`,
-    extendedAttributeReference(name) {
-      if (!extendedAttributesLinks.has(name)) {
-        return hyperHTML`<a>${name}</a>`;
-      }
-      return hyperHTML`<a data-cite="${extendedAttributesLinks.get(
-        name
-      )}">${name}</a>`;
-    },
-  };
-  const result = webidl2.write(parse, { templates });
-  return hyperHTML`<pre class="def idl">${result}</pre>`;
-}
+const templates = {
+  wrap(items) {
+    return items
+      .reduce(flatten, [])
+      .filter(x => x !== "")
```

Comment:
@saschanaz, wrap gets quite a few empty "", maybe a small bug/nit in Webidl2.js or are they somehow meaningful?

Reply:
webidl2 currently doesn't check whether a token has an empty whitespace or not and emits it unconditionally. Maybe we could fix it.

Reply:
I don't mind handling them here for now.